packages/grafana-data/src/types/vector.ts

```diff
@@ -1,35 +1,42 @@
-export interface Vector<T = any> {
-  length: number;
+export abstract class Vector<T = any> {
+  // Default iterator, but can be overridden
+  *[Symbol.iterator](): IterableIterator<T> {
+    for (let i = 0; i < this.length; i++) {
+      yield this.get(i);
+    }
+  }
+
+  abstract length: number;
 
   /**
    * Access the value by index (Like an array)
    */
-  get(index: number): T;
+  abstract get(index: number): T;
 
   /**
-   * Get the resutls as an array.
+   * Get the results as an array.
    */
-  toArray(): T[];
+  abstract toArray(): T[];
```

Comment:
Could probably have a default implementation for this as well. Maybe something for another PR.